src/crawlee/errors.py

```diff
@@ -39,13 +39,19 @@ class ProxyError(SessionError):
 class HttpStatusCodeError(Exception):
     """Raised when the response status code indicates an error."""
 
+    def __init__(self, message: str, status_code: int) -> None:
+        self.status_code = status_code
+        self.message = message
+        super().__init__(f'{message} (status code: {status_code}).')
```

Comment:
Might be safer to call this on the first line so that it doesn't overwrite something we set before... Unlikely to be a problem here though.